File: Adhiti_Shaili/Day 17/Two Sum.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        unordered_map<int,int> hash;
        vector<int> ans;
        for(int i=0;i<n;i++)
        {
            if(hash[target-nums[i]])
            {
                ans.push_back(hash[target-nums[i]]-1);
                ans.push_back(i);
                return ans;
            }
            hash[nums[i]]=i+1;
        }
        return ans;
    }
};

File: parnamondal/Week -4/Day -1/Two Sum.py
# Time Complexity - O(N)
# Space Complexity - O(N)

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

                        
        ma={}
        for i in range(len(nums)):
            ma[nums[i]]=i
        for i in range(len(nums)):
            ok=target -nums[i]
            if ok in ma and ma[ok]!=i:
                return [i,ma[ok]]
                
                